host/xtest/pkcs11_1000.c

```diff
@@ -8016,3 +8016,442 @@ static void xtest_pkcs11_test_1025(ADBG_Case_t *c)
+	};
+	CK_MECHANISM rsa_aes_key_wrap_mechanism = {
+		CKM_RSA_AES_KEY_WRAP, &rsa_aes_params, sizeof(rsa_aes_params),
+	};
+
+	uint8_t unwrapped_val[RSA_AES_MAX_KEY_SIZE] = { 0 };
+	CK_ULONG unwrapped_key_len = 0;
+	CK_ATTRIBUTE get_template_unwrapped[] = {
+		{ CKA_VALUE_LEN, &unwrapped_key_len,
+		  sizeof(unwrapped_key_len) },
+		{ CKA_VALUE, unwrapped_val, sizeof(unwrapped_val) },
+	};
+
+	CK_ULONG i = 0;
+
+	CK_ATTRIBUTE private_key_template[] = {
+		{ CKA_CLASS, &(CK_OBJECT_CLASS){ CKO_PRIVATE_KEY },
+		  sizeof(CK_OBJECT_CLASS) },
+		{ CKA_KEY_TYPE,	&(CK_KEY_TYPE){ CKK_RSA }, sizeof(CK_KEY_TYPE) },
+		{ CKA_TOKEN, &(CK_BBOOL){ CK_TRUE }, sizeof(CK_BBOOL) },
+		{ CKA_SIGN, &(CK_BBOOL){ CK_TRUE }, sizeof(CK_BBOOL) },
+		{ CKA_DECRYPT, &(CK_BBOOL){ CK_TRUE }, sizeof(CK_BBOOL) },
+		{ CKA_SENSITIVE, &(CK_BBOOL){ CK_TRUE }, sizeof(CK_BBOOL) },
+		{ CKA_EXTRACTABLE, &(CK_BBOOL){ CK_TRUE }, sizeof(CK_BBOOL) },
+		{ CKA_UNWRAP, &(CK_BBOOL){ CK_TRUE }, sizeof(CK_BBOOL) },
+		{ CKA_MODULUS, (CK_VOID_PTR)ac_rsassa_vect2_modulus,
+		  sizeof(ac_rsassa_vect2_modulus) },
+		{ CKA_PRIVATE_EXPONENT, (CK_VOID_PTR)ac_rsassa_vect2_priv_exp,
+		  sizeof(ac_rsassa_vect2_priv_exp) },
+		{ CKA_PUBLIC_EXPONENT, (CK_VOID_PTR)ac_rsassa_vect2_pub_exp,
+		  sizeof(ac_rsassa_vect2_pub_exp) },
+		{ CKA_PRIME_1, (CK_VOID_PTR)ac_rsassa_vect2_prime1,
+		  sizeof(ac_rsassa_vect2_prime1) },
+		{ CKA_PRIME_2, (CK_VOID_PTR)ac_rsassa_vect2_prime2,
+		  sizeof(ac_rsassa_vect2_prime2) },
+	};
+
+
+	rv = init_lib_and_find_token_slot(&slot);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		return;
+
+	rv = init_test_token(slot);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto close_lib;
+
+	rv = init_user_test_token(slot);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto close_lib;
+
+	rv = C_OpenSession(slot, session_flags, NULL, 0, &session);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto close_lib;
+
+	rv = C_Login(session, CKU_USER,	test_token_user_pin,
+		     sizeof(test_token_user_pin));
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto close_session;
+
+	for (i = 0; i < ARRAY_SIZE(rsa_aes_wrap_tests); i++) {
+		rv = test_rsa_aes_wrap(c, session, &rsa_aes_wrap_tests[i]);
+		if (rv != CKR_OK)
+			goto logout;
+	}
+
+	Do_ADBG_BeginSubCase(c, "Test external key unwrap with RSA AES");
+	rv = C_CreateObject(session, private_key_template,
+			    ARRAY_SIZE(private_key_template), &private_key);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	rv = C_UnwrapKey(session, &rsa_aes_key_wrap_mechanism, private_key,
+			 (CK_VOID_PTR)pkcs11_rsa_aes_wrapped_key,
+			 ARRAY_SIZE(pkcs11_rsa_aes_wrapped_key), key_template,
+			 ARRAY_SIZE(key_template), &unwrapped_key);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	rv = C_GetAttributeValue(session, unwrapped_key, get_template_unwrapped,
+				 ARRAY_SIZE(get_template_unwrapped));
+	if (!ADBG_EXPECT_CK_OK(c, rv) ||
+	    !ADBG_EXPECT_BUFFER(c, unwrapped_val, unwrapped_key_len,
+				pkcs11_rsa_aes_tagret_key,
+				ARRAY_SIZE(pkcs11_rsa_aes_tagret_key)))
+		goto out;
+out:
```

Comment:
these 2 lines are valid but looks a bit weird.

Reply:
removed in [a4f3f71](https://github.com/OP-TEE/optee_test/pull/627/commits/a4f3f7118946a5f2be78e74c1bc82c65445f9752)